phy/scripts/phy_script.py

```diff
@@ -32,169 +39,250 @@ def error(self, message):
+        p.add_argument('num_starting_clusters', type=int,
+                       # dest='num_starting_clusters',
+                       help='initial number of clusters',
+                       )
+        p.add_argument('--clustering', default='main',
+                       help='name of the clustering to use')
+        p.set_defaults(func=cluster_auto)
+
+    def create_notebook(self):
+        # TODO
+        pass
+
+    def parse(self, args):
+        return self._parser.parse_args(args)
 
-    parser.add_argument('--line-profiler', '-lp',
-                        dest='line_profiler',
-                        action='store_true',
-                        help='activate the line-profiler -- you need to '
-                        'decorate the functions to profile with `@profile` '
-                        'in the code')
 
-    parser.add_argument('--ipython', '-i', action='store_true',
-                        help='launch the script in an interactive '
-                        'IPython console')
+#------------------------------------------------------------------------------
+# Subcommand functions
+#------------------------------------------------------------------------------
 
-    parser.add_argument('--clustering', default='main',
-                        help='name of the clustering to use')
+def _get_kwik_path(args):
+    kwik_path = args.file
 
-    parser.add_argument('--cluster_ids', '-c',
-                        help='list of clusters to select initially')
+    if not op.exists(kwik_path):
+        print("The file `{}` doesn't exist.".format(kwik_path))
+        return
 
-    parse, extra = parser.parse_known_args(args)
-    kwargs = _parse_extra(extra)
-    return parse, kwargs
+    return kwik_path
 
 
-def run_manual(kwik_path, clustering=None, interactive=False,
-               cluster_ids=None):
-    import phy
+def _create_session(args, **kwargs):
     from phy.cluster import Session
-    from phy.gui import start_qt_app, run_qt_app
+    kwik_path = _get_kwik_path(args)
+    session = Session(kwik_path, **kwargs)
+    return session
 
-    if not op.exists(kwik_path):
-        print("The file `{}` doesn't exist.".format(kwik_path))
-        return 1
 
-    print("\nLoading {}...".format(kwik_path))
-    session = Session(kwik_path,
-                      clustering=clustering,
-                      )
-    print("Data successfully loaded!\n")
+def describe(args):
+    from phy.io.kwik import KwikModel
+    path = _get_kwik_path(args)
+    model = KwikModel(path, clustering=args.clustering)
+    return 'model.describe()', dict(model=model)
+
+
+def traces(args):
+    from vispy.app import run
+    from phy.plot.traces import TraceView
+    from phy.io.h5 import open_h5
+    from phy.io.traces import read_kwd, read_dat
+
+    path = args.file
+    if path.endswith('.kwd'):
+        f = open_h5(args.file)
+        traces = read_kwd(f)
+    elif path.endswith(('.dat', '.bin')):
```

Comment:
there should be an else: "filetype not supported, please rename /..."

also we should make sure we're not trying to read/write to the same KWD file...!